eng/workloads/workloads.csproj

```diff
@@ -0,0 +1,205 @@
+    <AutoGenerateAssemblyVersion>true</AutoGenerateAssemblyVersion>
+    <DotNetUseShippingVersion>true</DotNetUseShippingVersion>
+    <!-- Temp directory for light command layouts -->
+    <LightCommandObjDir>$(ArtifactsObjDir)/LightCommandPackages</LightCommandObjDir>
+    <!-- Directory for the zipped up light command package -->
+    <LightCommandPackagesDir>$(ArtifactsNonShippingPackagesDir)</LightCommandPackagesDir>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <WixToolsetPath>$(PkgMicrosoft_Signed_Wix)\tools</WixToolsetPath>
+    <SwixPluginPath>$(PkgMicroBuild_Plugins_SwixBuild_Dotnet)</SwixPluginPath>
+    <SwixBuildTargets>$(SwixPluginPath)\build\MicroBuild.Plugins.SwixBuild.targets</SwixBuildTargets>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Workloads" Version="$(MicrosoftDotNetBuildTasksWorkloadsVersion)" GeneratePathProperty="true" />
+    <PackageReference Include="Microsoft.Signed.Wix" Version="$(WixPackageVersion)" GeneratePathProperty="true" />
+    <PackageReference Include="MicroBuild.Plugins.SwixBuild.Dotnet" Version="$(SwixPackageVersion)" GeneratePathProperty="true" />
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Installers" Version="$(MicrosoftDotNetBuildTasksInstallersVersion)" GeneratePathProperty="True" />
+  </ItemGroup>
+
+  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
+
+  <Target Name="Build" DependsOnTargets="GetAssemblyVersion;_GetVersionProps;_GenerateMsiVersionString">
+    <ItemGroup>
+      <!-- Visual Studio components must be versioned. Build tasks will fall back to cobbling a version number from
+        the manifest information unless it's overridden. -->
+      <!-- Overrides for Visual Studio setup generation. If the workload definition IDs change,
+            these must be updated. -->
+      <ComponentResources Include="microsoft-net-sdk-wasi"
+                          Version="$(FileVersion)"
+                          Title=".NET Wasi SDK"
+                          Description="Wasi SDK and Build tools for .NET"/>
+    </ItemGroup>
+
+    <!-- Shorten package names to avoid long path issues in Visual Studio -->
+    <ItemGroup>
+      <ShortNames Include="microsoft.net.workload.wasi.net;Microsoft.NET.Workload.Wasi.net;microsoft.net.workload.wasi.current">
+        <Replacement>Wasi.</Replacement>
+      </ShortNames>
+      <ShortNames Include="microsoft.netcore.app.runtime;Microsoft.NETCore.App.Runtime;microsoft.net.runtime;Microsoft.NET.Runtime">
+        <Replacement>Microsoft</Replacement>
+      </ShortNames>
+    </ItemGroup>
+
+    <ItemGroup>
+      <ManifestPackages Include="$(PackageSource)Microsoft.NET.Workload.Wasi*.Manifest-%(WorkloadSdkBandVersions.Identity)*.*.nupkg"
+                        Exclude="$(PackageSource)Microsoft.NET.Workload.Wasi*.Manifest*.msi*.nupkg"
+                        MsiVersion="$(MsiVersion)"
+                        SupportsMachineArch="%(SupportsMachineArch)" />
```

Comment:
The `SupportsMachineArch` metadata determines whether you plan on support arm64 authoring in VS. If you don't have arm64 support, you can set this to false or otherwise, alias the pack for win-arm64 in the workload manifest and use the x64 binaries and let it run under emulation. We did this for some of the EMSDK pieces early during arm64 support in VS. But right now, we cannot insert this into VS because it's looking for an arm64 installer that doesn't exist.

Reply:
I'm not sure what the problem is because we are producing arm64 packs for wasi.